scripts/import/import_clinvar_xml

```diff
@@ -1046,19 +1050,73 @@ sub get_variant{
       foreach my $vf (@$var_feats){
         warn "vf:". $vf->name . " allele string: ". $vf->allele_string . "\n";
       }
+    } else {
+      warn "no variation feature found via search by location identifier: $location_str\n" if $DEBUG ==1;
     }
-    else{
-      ## try to match to multi-allelic variants by location
-      return get_variant_by_slice_allele($record, $alt_allele);
-    }
-  }
-  else {
-    warn "No ref or alt allele or location found\n" if $DEBUG == 1;
-    return undef;
+  } else {
+    warn "no ref or alt or chr or start, skipping lookup by location identifier\n";
   }
+
+  #try to match by SPDI: longest allele format
+  #returns undef for ref==alt SPDIs
+  my $result = get_variant_via_SPDI ($record, $alt_allele);
```

Comment:
Did this work in your testing?

get_variant_via_SPDI returns an object, an array of objects and a scalar, but only the scalar is caught here. I get an error message:
ERROR: Can't call method "add_evidence_value" without a package or object reference at scripts/import/import_clinvar_xml line 753.

and it's calling  add_evidence_value on the alt allele string '-'

Reply:
I haven't seen that error before. What xml did you test? I can have a look.

Reply:
I'm using RCV000001471 - I've put the XML in the ClinVar/testdata/ directory on nfs.